Prevent potential XXE vulnerability in modRestService by disabling libxml entity loader

Mark-H opened the pull request:
### What does it do?

Disables the libxml entity loader when reading an XML string in a request to a modRestService API. This is enabled by default, but can be disabled with a service option if custom DTDs are necessary in an API.

### Why is it needed?

Makes modRestService implementations more secure by default, fixing the report in #15237. The MODX core does not use this particular service, so only third party implementations would be affected. 

### Related issue(s)/PR(s)
 
Fixes #15237

Jako reviewed (commented):
What about this: https://php.watch/versions/8.0/libxml_disable_entity_loader-deprecation

BTW libxml 2.9.0 should be available since 2012.

opengeek commented:
> What about this: https://php.watch/versions/8.0/libxml_disable_entity_loader-deprecation
> 
> BTW libxml 2.9.0 should be available since 2012.

Interesting, @Jako. We should probably add one of those recommended checks in there to avoid future deprecated warning issues.

Mark-H committed:
Prevent potential XXE vulnerability in modRestService by disabling the libxml entity loader [#15237]

The libxml_disable_entity_loader function is deprecated in PHP8, and the entity loader is automatically enabled on v2.9.0+ of libxml which may have been used pre-PHP8 as well. PHP8 comes with at least v2.9.0+ of libxml bundled, so this conditional covers both scenarios.

Ref: https://github.com/php/php-src/pull/5867

Mark-H commented:
Nice catch, I hadn't seen that yet. Updated to check the `LIBXML_VERSION` constant, which was recommended in the PR on php-src that deprecated it as it's supposedly possible for builds to already use a newer version that's safe by default pre-PHP8.

JoshuaLuckers commented on core/model/modx/rest/modrestservice.class.php:
```diff
@@ -59,6 +59,7 @@ public function __construct(modX &$modx,array $config = array()) {
             'responseSuccessKey' => 'success',
             'trimParameters' => false,
             'xmlRootNode' => 'response',
+            'xmlDisableEntityLoader' => true,
```
Since the setting only works for a specific version range of the library, do we need this as default setting?
